### PR TITLE
Add GitHub Actions for lint

### PR DESCRIPTION
**Pull request by yihong0618**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -199,3 +199,8 @@ meta = gp.GPTableMeta(table_output_name, table_output_schema, columns_output_typ
 GPDatabase_instance.check_table_if_exist(table_name, schema)
 
 ```
+## PR
+
+- install dev requirements `pip install -r requirements-dev.txt`
+- `black` the code use `black .  --line-length 100`
```
Comment (beeender) [inline README.md:205]: Doesn't black have a config file so we can put it into the root of project?
I really need my editor be aware of the `100` limit

Comment (yihong0618) [reply README.md:205]: Yes it has config using `toml` https://black.readthedocs.io/en/stable/usage_and_configuration/the_basics.html

For now, only one rule I think its not necessary to add new file~

Comment (beeender) [reply README.md:205]: please add it, I really need the lint config to make my editor happy. Thanks!

Comment (yihong0618) [reply README.md:205]: added

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        with:
+          path: ${{ steps.set_variables.outputs.PIP_CACHE }}
+          key: ${{ runner.os }}-pip-${{ steps.set_variables.outputs.PY }}
+      - name: Install dependencies
+        run:  |
+          python -m pip install --upgrade pip
+          pip install -r requirements-dev.txt
+      - name: Run black lint
+        run:  | 
+          black .  --line-length 100
```
Comment (beeender) [inline .github/workflows/ci.yml:41]: Even with the cfg file, we still need to pass the argument here?

Comment (yihong0618) [reply .github/workflows/ci.yml:41]: yes...for command line `black` needs it.

Comment (yihong0618) [reply .github/workflows/ci.yml:41]: no need, (wo shabi le)

Comment (yihong0618) [reply .github/workflows/ci.yml]: fixed.....

**Review by beeender — APPROVED** [review]

**Pull request merged**